### PR TITLE
Fix IsDockerWorking class not using TestContainersStrategy

### LINKED ISSUE #29188: IsDockerWorking not using TestContainersStrategy after 2.13.4.Final
### Describe the bug

There is a regression in the class `IsDockerWorking`, it was detected running on GitLab CI using the docker socket configuration: https://www.testcontainers.org/supported_docker_environment/continuous_integration/gitlab_ci/

The first strategy should be using the `TestContainersStrategy` and is able to detect the socket correctly.

I tested it with and without a `~/.testcontainers.properties`, but it's the same behavior.

The only related change is https://github.com/quarkusio/quarkus/pull/28702

cc: @geoand 

### Expected behavior

Running Quarkus 2.13.3.Final:

```
2022-11-10 18:36:35,116 DEBUG [io.qua.dep.IsDockerWorking] (build-40) Checking Docker Environment using strategy io.quarkus.deployment.IsDockerWorking$TestContainersStrategy
2022-11-10 18:36:35,161 INFO  [org.tes.doc.DockerClientProviderStrategy] (build-40) Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
2022-11-10 18:36:36,001 INFO  [org.tes.doc.DockerClientProviderStrategy] (build-40) Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
```
Only the TestContainersStrategy runs and works correctly.

### Actual behavior

Running Quarkus 2.13.3.Final and 2.14.0.Final:

On the docker.sock mounted CI pipeline failing:
```
2022-11-10 17:19:04,262 DEBUG [io.qua.dep.IsDockerWorking] (build-8) Checking Docker Environment using strategy io.quarkus.deployment.IsDockerWorking$TestContainersStrategy
2022-11-10 17:19:04,285 DEBUG [io.qua.dep.IsDockerWorking] (build-8) Checking Docker Environment using strategy io.quarkus.deployment.IsDockerWorking$DockerHostStrategy
2022-11-10 17:19:04,286 DEBUG [io.qua.dep.IsDockerWorking] (build-8) Checking Docker Environment using strategy io.quarkus.deployment.IsDockerWorking$DockerBinaryStrategy
2022-11-10 17:19:04,321 WARN  [io.qua.dep.IsDockerWorking] (build-8) No docker binary found or general error: java.lang.RuntimeException: Input/Output error while executing command.
```

On my local laptop fallback to using the docker binary detection:
```
2022-11-10 18:21:17,864 DEBUG [io.qua.dep.IsDockerWorking] (build-36) Checking Docker Environment using strategy io.quarkus.deployment.IsDockerWorking$TestContainersStrategy
2022-11-10 18:21:17,886 DEBUG [io.qua.dep.IsDockerWorking] (build-36) Checking Docker Environment using strategy io.quarkus.deployment.IsDockerWorking$DockerHostStrategy
2022-11-10 18:21:17,887 DEBUG [io.qua.dep.IsDockerWorking] (build-36) Checking Docker Environment using strategy io.quarkus.deployment.IsDockerWorking$DockerBinaryStrategy
2022-11-10 18:21:17,969 DEBUG [io.qua.dep.IsDockerWorking] (build-36) Docker daemon found. Version: '20.10.21'
2022-11-10 18:21:17,987 INFO  [org.tes.uti.ImageNameSubstitutor] (build-36) Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
2022-11-10 18:21:18,007 INFO  [org.tes.doc.DockerClientProviderStrategy] (build-36) Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
2022-11-10 18:21:18,816 INFO  [org.tes.doc.DockerClientProviderStrategy] (build-36) Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
```

It runs the strategies TestContainersStrategy, DockerHostStrategy, DockerBinaryStrategy... here it works since it find the docker binary, yet TestContainers is using the UnixSocketClientProviderStrategy.

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

openjdk 17.0.5 2022-10-18

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.13.4.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.6

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by jorsol**
Fixes #29188

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/IsDockerWorking.java [hunk 1]
```diff
@@ -84,7 +84,7 @@ public Result get() {
 
                 Class<?> configurationClass = Thread.currentThread().getContextClassLoader()
                         .loadClass("org.testcontainers.utility.TestcontainersConfiguration");
-                Object configurationInstance = configurationClass.getMethod("instance").invoke(null);
+                Object configurationInstance = configurationClass.getMethod("getInstance").invoke(null);
```
Comment (geoand) [inline core/deployment/src/main/java/io/quarkus/deployment/IsDockerWorking.java:87]: Great catch!

File: core/deployment/src/main/java/io/quarkus/deployment/DockerStatusProcessor.java [hunk 2]
```diff
@@ -8,6 +8,6 @@ public class DockerStatusProcessor {
 
     @BuildStep
     DockerStatusBuildItem IsDockerWorking(LaunchModeBuildItem launchMode) {
-        return new DockerStatusBuildItem(new IsDockerWorking(launchMode.getLaunchMode().isDevOrTest()));
+        return new DockerStatusBuildItem(new IsDockerWorking(!launchMode.getLaunchMode().isDevOrTest()));
```
[the remark this answers is not in the document]
Comment (jorsol) [reply core/deployment/src/main/java/io/quarkus/deployment/DockerStatusProcessor.java:11]: The IsDockerWorking has a "silent" parameter, in Dev/Test mode this is true, and in prod is false, honestly, I expect it to be silent on Prod, and not on Dev/Test mode.

So correct me if I'm wrong but the parameter should show the debug when something is failing, and on Dev/Test mode was not triggered.
https://github.com/quarkusio/quarkus/blob/a40259783667f4aad40316baff120726da6c3450/core/deployment/src/main/java/io/quarkus/deployment/IsDockerWorking.java#L105-L108

Anyway, if you think this is not right I could revert it.

[the remark this answers is not in the document]
Comment (geoand) [reply core/deployment/src/main/java/io/quarkus/deployment/DockerStatusProcessor.java:11]: This code is never run in prod mode, so I don't really know why it's done the way it is.

In any case, let's revert this change and focus on the actual fix

**Review by geoand — APPROVED** [review]
Thanks!

**Comment by jorsol** [comment]
What about also `triage/backport-2.13? ` since it affects that version too?

**Comment by geoand** [comment]
It does? Are you sure?

**Comment by jorsol** [comment]
> It does? Are you sure?

Yes, I'm sure, it was introduced into `2.13.4.Final`, that's where I caught this: https://github.com/quarkusio/quarkus/pull/28702

**Comment by geoand** [comment]
Okay, thanks for checking

**Pull request merged**